addon/index.ts

```diff
@@ -11,30 +12,33 @@ function resolveInitializer(moduleName) {
   return initializer;
 }
 
-function registerInitializers(app, moduleNames) {
+function registerInitializers(app: typeof Engine, moduleNames: string[]) {
```

Comment:
Can we make this `app: import('@ember/engine')` and remove the top level import?

Reply:
Not sure if it's possible. Looks like `typeof import('@ember/engine')` is a `Promise`, that makes the following line fail:
```
app.initializer(resolveInitializer(moduleNames[i]));
```
with

> Property 'initializer' does not exist on type 'typeof import('...'.

Reply:
Hmm, that `import` syntax is specifically a type annotation only (which typescript is aware of), and shouldn't impact the runtime at all.

Reply:
Also, I'm not sure why you need `typeof import('@ember/engine')`, I would have thought:

```suggestion
function registerInitializers(app: import('@ember/engine'), moduleNames: string[]) {
```

Would be all you would need...

Reply:
Hm.. with `app: import('@ember/engine')` I get the following error:

> Module '@ember/engine' does not refer to a type, but is used as a type here. Did you mean 'typeof import('@ember/engine')'?

When I change it to `app: import('@ember/engine').default` it complains in `   app.initializer(resolveInitializer(moduleNames[i]));` with

> Property 'initializer' is a static member of type 'Engine'

The best I could do, so there are no type errors in a module, is adding a `typeof` again, like:

```
export default function loadInitializers(app: typeof import('@ember/engine').default, prefix: string): void {
```

Which is verbose and also it has to be changed in 2 other function signatures.

Reply:
> I'm not sure why you need typeof import('@ember/engine')

This comes from the typings where `Engine.initializer` is static:

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/8a8bc934a547e3a3c25449fdd2dc0f410d173723/types/ember__engine/index.d.ts#L27

Reply:
Seems like a bug 😄

Reply:
https://github.com/emberjs/ember.js/blob/6449d30026e673736e91f3ebe62fbf03f8b09b31/packages/%40ember/engine/index.js#L105-L115 (both `initializer` and `instanceInitializer` exist on the instance

Reply:
But I guess in this case, we actually do get the class instead 🤔 (since thats the API of this lib)

Reply:
Maybe just change it to `interface EngineLike { initializer(): stuff; }`?

Reply:
Thank for explanation. Yes, even after fixing typings upstream it still sounds like a good idea to have local "valid" interface to not care about required `@ember/engines` typings version..

Reply:
I've tried to change `initializer(` and `instanceInitializer(` from `static` to instance methods in DefinitelyTyped, but `@ember/engine` types tests currently [clearly assume](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/518700405e6f24d00288c4bb3bf56d2d4b8dae97/types/ember__engine/test/engine.ts#L4-L8) these are static methods.
Also, according to EmberObserver CodeSearch, there are quite a lots of uses of that `initializer` as a static method: https://github.com/emberjs/data/blob/39182f324d5777e4a75c4a582f0791631e37d8de/packages/-ember-data/tests/integration/application-test.js#L100-L101

However, there are few examples of instance method usages https://github.com/emberjs/ember-test-helpers/blob/4ca1ad848e8346865c14e0e498f8a35f7a6cbe48/tests/unit/setup-context-test.js#L294-L306

TBH, I'm not very familiar with Engines design, but if consider a regular app code snippet, like:

```js
const App = Application.extend({
  modulePrefix: config.modulePrefix,
  podModulePrefix: config.podModulePrefix,
  Resolver
});

loadInitializers(App, config.modulePrefix);
```

it looks like we pass an App class to the `loadInitializers` which means we should keep that methods(`initialize`, `instanceInitializer`) static.

@rwjblue I would appreciate your help. Should it be both static and an instance method at the same time or can we let it to stay static in this PR?

Reply:
ok, I can see that both kinds of invocations seem to be supported. 

Previously you've shared: https://github.com/emberjs/ember.js/blob/6449d30026e673736e91f3ebe62fbf03f8b09b31/packages/%40ember/engine/index.js#L105-L115 - which proves its instance method nature

But few lines below: https://github.com/emberjs/ember.js/blob/6449d30026e673736e91f3ebe62fbf03f8b09b31/packages/%40ember/engine/index.js#L304 - seems like we also add a class method..

However, at the moment I'm pretty sure for `ember-load-initializers` we only need static